CMakeLists.txt

```diff
@@ -621,3 +621,35 @@ if (SPIRV_CROSS_CLI)
 		endif()
 	endif()
 endif()
+
+if (NOT SPIRV_CROSS_SKIP_INSTALL)
+	file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-config.cmake.in" [=[
+	    @PACKAGE_INIT@
+	    include("${CMAKE_CURRENT_LIST_DIR}/spirv-cross-targets.cmake")
+	]=])
+
+	configure_package_config_file(
+	    "${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-config.cmake.in"
+	    "${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-config.cmake"
+	    INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/spirv-cross
+	)
+
+	write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/spirv-cross-config-version.cmake"
+		VERSION ${SPIRV_CROSS_VERSION}
```

Comment:
There is no SPIRV-Cross "version" per-se and I think the .so version is bumped when the C API is updated. Whatever shipped in an SDK shipped in an SDK, and I think someone tags those commits.

Maybe it would be possible for SDK builds to explicitly override a version string or something.

Reply:
Are you suggesting that we allow user to choose what would be the `SPIRV-Cross` version when configuring the project ? (e.g. `cmake .. -DSPIRV_CROSS_VERSION:STRING=1.3.250`)

> Using the git commit as a version is probably the only reasonable way if there is no tag for that particular commit.

Sounds reasonable, just to double check we would then have version which are not human readable as they would simply be git (short) SHA right ?
Ideally I would prefer that if we ever choose a version scheme a human readable one is picked, or better a semantic versioning (especially since you already chose to specify an ABI version this should be straightforward)

Reply:
I'd be fine with an option override for version strings. What is the intention for that version string? Package maintainers in distros and LunarG when packaging it in an SDK release?

Reply:
From my perspective I am interested for it as a package maintainer for `vcpkg` yes.

Reply:
Can we remove `SPIRV-CrossConfigVersion.cmake` from this pr and discuss this in an issue?
That'd unblock this pr, and solve the original intent of this pr.

Filtering on versions can be added in a later pr.